tools/projmgr/docs/Overview.md

```diff
@@ -0,0 +1,489 @@
+      - file:          # file name along with path
+        include:       # include group for a list of *build* and *target* types.
+        exclude:       # exclude group for a list of *build* and *target* types.
+        optimize:      # optimize level for code generation (optional)
+        debug:         # generation of debug information (optional)
+        defines:       # symbol and path settings for code generation (optional).
+        misc:          # Literal tool-specific controls
+```
+
+**Example:**
+
+Add source files to a project or a software layer.  Used in `*.cproject.yml` and `*.clayer.yml` files.
+
+```yml
+groups:
+  - group:  "Main File Group"
+    exclude: .Release~Virtual, .Test-DSP~Virtual, ~Board
+    files: 
+      - file: file1a.c
+      - file: file1b.c
+        defines:
+          - define: a=1
+          - undefine: b
+        optimize: size
+
+  - group: "Other File Group"
+    files:
+      - file: file2a.c
+        include: ~Virtual
+        defines: 
+          - define: test=2
+      - file: file2a.c
+        exclude: ~Virtual
+      - file: file2b.c
+
+  - group: "Nested Group"
+    groups:
+      - group: Subgroup1
+        files:
+          - file: file-sub1-1.c
+          - file: file-sub1-2.c
+      - group: Subgroup2
+        files:
+          - file: file-sub2-1.c
+          - file: file-sub2-2.c
+```
+
+## Layers 
+
+Add a software layer to a project.  Used in `*.cproject.yml` files.
+
+YML structure:
+```yml
+layers:                # Start a list of layers
+  - layer:             # path to the `*.clayer.yml` file that defines the layer (required).
+    include:           # include layer for a list of *build* and *target* types (optional).
+    exclude:           # exclude layer for a list of *build* and *target* types (optional).
+    optimize:          # optimize level for code generation (optional).
+    debug:             # generation of debug information (optional).
+    defines:           # symbol and path settings for code generation (optional).
+    misc:              # Literal tool-specific controls.
+```
+
+## Components
+
+Add software components to a project or a software layer.  Used in `*.cproject.yml` and `*.clayer.yml` files.
+
+YML structure:
+```yml
+components:            # Start a list of layers
+  - component:         # name of the software component.
+    include:           # include layer for a list of *build* and *target* types (optional).
+    exclude:           # exclude layer for a list of *build* and *target* types (optional).
+    optimize:          # optimize level for code generation (optional).
+    debug:             # generation of debug information (optional).
+    defines:           # symbol and path settings for code generation (optional).
+    misc:              # Literal tool-specific controls.
+```
+
+The proposed component name syntax is:
+`[Cvendor::] Cclass [&Cbundle] :Cgroup [:Csub] [&Cvariant] [@[>=]Cversion]`
```

Comment:
The `>=` operator has the drawback of potentially silently select incompatible versions, for example `@>=1.0` could select version `2.0` which is considered a major (potentially incompatible) update. Other tools make use of the `~=` operator to select _compatible_ versions, see [Python](https://www.python.org/dev/peps/pep-0440/#compatible-release) for instance. We could allow `@~=1.1` which means version between `1.1.0` and excluding `2.0.0`.